src/mame/seta/speglsht.cpp

```diff
@@ -228,68 +226,69 @@ void speglsht_state::cop_w(offs_t offset, uint32_t data, uint32_t mem_mask)
 {
 	COMBINE_DATA(&m_cop_ram[offset]);
 
-	if(m_cop_ram[offset]&0x8000) //fix (sign)
+	if (m_cop_ram[offset] & 0x8000) // 16 bit signed to 32 bit
 	{
-		m_cop_ram[offset]|=0xffff0000;
+		m_cop_ram[offset] |= 0xffff0000;
 	}
 }
 
 //matrix * vector
 uint32_t speglsht_state::cop_r(offs_t offset)
 {
-	int32_t *cop=(int32_t*)&m_cop_ram[0];
+	int32_t *cop = (int32_t*)&m_cop_ram[0];
 
 	union
 	{
 		int32_t  a;
 		uint32_t b;
-	}temp;
+	} temp;
 
 	switch (offset)
 	{
-		case 0x40/4:
+		case 0x40 / 4:
 		{
-			temp.a=((cop[0x3]*cop[0x0]+cop[0x4]*cop[0x1]+cop[0x5]*cop[0x2])>>14)+cop[0xc];
+			temp.a = ((cop[0x3] * cop[0x0] + cop[0x4] * cop[0x1] + cop[0x5] * cop[0x2]) >> 14) + cop[0xc];
 			return temp.b;
 		}
 
-		case 0x44/4:
+		case 0x44 / 4:
 		{
-			temp.a=((cop[0x6]*cop[0x0]+cop[0x7]*cop[0x1]+cop[0x8]*cop[0x2])>>14)+cop[0xd];
+			temp.a = ((cop[0x6] * cop[0x0] + cop[0x7] * cop[0x1] + cop[0x8] * cop[0x2]) >> 14) + cop[0xd];
 			return temp.b;
 		}
 
-		case 0x48/4:
+		case 0x48 / 4:
 		{
-			temp.a=((cop[0x9]*cop[0x0]+cop[0xa]*cop[0x1]+cop[0xb]*cop[0x2])>>14)+cop[0xe];
+			temp.a = ((cop[0x9] * cop[0x0] + cop[0xa] * cop[0x1] + cop[0xb] * cop[0x2]) >> 14) + cop[0xe];
 			return temp.b;
 		}
 	}
```

Comment:
This looks overly convoluted.  Can you reduce these by actually using the offset?  Like:
```c++
	switch (offset)
	{
	case 0x10:
	case 0x11:
	case 0x12:
		{
			unsigned displacement = (offset & 0x03) * 3;
			result = ((cop[3 + displacement] * cop[0] + cop[4 + displacement] * cop[1] + cop[5 + displacement] * cop[2]) >> 14) + cop[0xc];
			break;
		}
	}
```